### PR TITLE
Fix encode/decode failure when modifying open interval inside exclusion

### LINKED ISSUE #422: Internal error. Failed encode / decode check. on modify
Version: 1.4.2
OS: MACOS 10.15.7

This seems somewhat related to #416 

`timew start TRACKER-6145`
```
Tracking "TRACKER-6145"
  Started 2021-05-11T12:12:43
  Current                  43
  Total               0:00:00
```
`timew modify start @1 12:02`
```
Internal error. Failed encode / decode check.
```

`timew modify start @1 12:02 :debug`
```CLI Parser
  _original_args
    timew modify start @1 12:02 :debug
  _args
    word basename='timew' raw='timew' BINARY 
    word canonical='modify' raw='modify' ORIGINAL CMD 
    word raw='start' ORIGINAL FILTER TAG 
    word raw='@1' value='1' ORIGINAL ID FILTER 
    date raw='12:02' ORIGINAL FILTER 
    word canonical=':debug' raw=':debug' ORIGINAL HINT FILTER 

>> 2021-05.data: 14 intervals
>> Found 0 holidays
>> Found 7 exclusions
>> 2021-05.data: Deleted inc 20210511T161243Z # "TRACKER-6145"
>> 2021-05.data: Added inc 20210511T161243Z # "TRACKER-6145"
>> Found 0 holidays
>> Found 7 exclusions
>> 2021-05.data: Deleted inc 20210511T161243Z # "TRACKER-6145"
>> Loaded 0 tracked intervals
>> Datafile::addInterval() failed.
>> Encode / decode check failed:
>>   inc 20210511T160200Z # "TRACKER-6145"
>> is not equal to:
>>   inc 20210511T160200Z # "TRACKER-6145"
Internal error. Failed encode / decode check.
>> Timer timew 0.004484 sec
```

### PR DESCRIPTION
**Pull request by sruffell**
The fundamental change is to prevent marking the latest "real" interval as a synthetic interval in this case. 

Closes #422

### CODE REVIEW

**Comment by sruffell** [comment]
I pushed another commit onto the end of this series that simplifies the implementation of getIntervalsById and getTracked.  Arguably it should be another pull request, but it was bothering me when I was looking at this that one of the implementations did the right thing and the other did not.  Hopefully now someone looking at them can more easily see the differences in the implementation.

**Review by lauft — APPROVED** [review]
LGTM. Thanks for fixing.

File: test/modify.t [hunk 1]
```diff
@@ -249,6 +249,35 @@ class TestModify(TestCase):
...
 
+    def test_modify_move_start_inside_exclusion(self):
+        """`timew modify` should handle moving start times within an exclusion."""
+
+        now = datetime.now().replace(second=0, microsecond=0, minute=0)
+        three_hours_before = now - timedelta(hours=3)
+        four_hours_before = now - timedelta(hours=4)
+
+        now_utc = now.utcnow().replace(second=0, microsecond=0, minute=0)
+        day_before = now_utc - timedelta(days=1)
```
[the remark this answers is not in the document]
Comment (sruffell) [reply test/modify.t:260]: Addressed in #424

File: test/modify.t [hunk 2]
```diff
@@ -249,6 +249,35 @@ class TestModify(TestCase):
...
+        three_hours_before_utc = now_utc - timedelta(hours=3)
+        four_hours_before_utc = now_utc - timedelta(hours=4)
+        five_hours_before_utc = now_utc - timedelta(hours=5)
```
[the remark this answers is not in the document]
Comment (sruffell) [reply test/modify.t:263]: Addressed in #424.

**Pull request merged**